packages/evm-predicates/predicate/src/main.sw

```diff
@@ -32,21 +38,23 @@ struct SignedData {
 
 configurable {
     /// The Ethereum address that signed the transaction.
-    SIGNER: b256 = ZERO_B256,
+    SIGNER: b256 = b256::zero(),
 }
 
 fn main(witness_index: u64) -> bool {
     // Retrieve the Ethereum signature from the witness data in the Tx at the specified index.
-    let signature: B512 = tx_witness_data(witness_index).unwrap();
+    let witness_signature: B512 = tx_witness_data(witness_index).unwrap();
 
     // Hash the Fuel Tx (as the signed message) and attempt to recover the signer from the signature.
-    let result = ec_recover_evm_address(signature, personal_sign_hash(tx_id()));
+    let signature = Signature::Secp256k1(Secp256k1::from(witness_signature));
+    let pub_key = PublicKey::from(SIGNER);
+    let message = Message::from(personal_sign_hash(tx_id()));
+
+    let result = signature.verify(pub_key, message);
 
     // If the signers match then the predicate has validated the Tx.
     if result.is_ok() {
-        if SIGNER == result.unwrap().into() {
-            return true;
-        }
+        return true;
```

Comment:
Replacement of `ec_recover_evm_address` requires a double-check.

Reply:
this can be fixed on another PR as we won't generate predicate versions now